src/libsystemd-network/dhcp6-option.c

```diff
@@ -79,6 +79,52 @@ int dhcp6_option_append(uint8_t **buf, size_t *buflen, uint16_t code,
         return 0;
 }
 
+int dhcp6_option_append_vendor_option(uint8_t **buf, size_t *buflen, OrderedHashmap *vendor_options) {
+        uint32_t enterprise_identifier;
+        sd_dhcp6_option *options;
+        size_t total, offset;
+        Iterator i;
+        int r;
+
+        assert(buf);
+        assert(*buf);
+        assert(buflen);
+        assert(vendor_options);
+
+        ORDERED_HASHMAP_FOREACH(options, vendor_options, i) {
+                _cleanup_free_ uint8_t *p = NULL;
+                uint8_t *q;
+
+                enterprise_identifier = htobe32(SYSTEMD_PEN);
+
+                p = memdup(&enterprise_identifier, sizeof(enterprise_identifier));
+                if (!p)
+                        return -ENOMEM;
+
+                total = sizeof(enterprise_identifier);
+                offset = total;
```

Comment:
I fail to see the point of `total` and `offset` when both of them are always `sizeof(enterprise_identifier)`. Wouldn't one, in the foreach scope, be sufficient?

Reply:
what is difficult is this. why mixup offset and total ?